elasticdl/python/tests/optimizer_wrapper_test.py

```diff
@@ -535,8 +569,189 @@ def _test_correctness(self, optimizer_class, **kwargs):
                 for w1, w2 in zip(layer1.weights, layer2.weights):
                     self.assertTrue((w1 - w2 < 0.0001).numpy().all())
 
-    def test_correctness(self):
-        self._test_correctness(SGD, learning_rate=0.1, momentum=0.5)
+    def test_sgd_correctness(self):
+        seed = 1
+        X, Y = _prepare_random_data(
+            iters_per_epoch=4,
+            batch_size=4,
+            input_length=6,
+            input_dim=4,
+            is_sparse=True,
+            random_seed=seed,
+        )
+        self._test_correctness(SGD, X, Y, seed, learning_rate=0.1)
+        self._test_correctness(
+            SGD, X, Y, seed, learning_rate=0.1, momentum=0.5
+        )
+
+    def test_adadelta_correctness(self):
+        seed = 1
+        X, Y = _prepare_random_data(
```

Comment:
Can we reuse this across tests and then maybe write a loop that iterates through different optimizers, finally check `self._test_correctness`? There seems to be a lot of duplicated code here.

Reply:
Yes! Done.